server/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -1200,9 +1203,31 @@ public IndicesService getIndicesService() {
         return indicesService;
     }
 
-    public InternalAggregation.ReduceContext createReduceContext(boolean finalReduce) {
-        return new InternalAggregation.ReduceContext(bigArrays, scriptService,
-            finalReduce ? multiBucketConsumerService.create() : bucketCount -> {}, finalReduce);
+    /**
+     * Returns a builder for {@link InternalAggregation.ReduceContext}. This
+     * builder retains a reference to the provided {@link SearchRequest}.
+     */
```

Comment:
@jimczi, I remember us trying not to hold on to references to the `SearchRequest` because it could be big. Or something like that. Is that still a thing? It looks like we keep the `SearchRequest` around for a while during the search right now.